test/mitmproxy/net/test_tcp.py

```diff
@@ -421,16 +421,18 @@ class TestServerCipherList(tservers.ServerTestBase):
         cipher_list='AES256-GCM-SHA384'
     )
 
+    @pytest.mark.xfail
     def test_echo(self):
+        # Not working for OpenSSL 1.1.1, see
+        # https://github.com/pyca/pyopenssl/blob/fc802df5c10f0d1cd9749c94887d652fa26db6fb/src/OpenSSL/SSL.py#L1192-L1196
         c = tcp.TCPClient(("127.0.0.1", self.port))
         with c.connect():
             c.convert_to_tls(sni="foo.com")
-            expected = b"['AES256-GCM-SHA384']"
-            assert c.rfile.read(len(expected) + 2) == expected
+            expected = b"['TLS_AES_256_GCM_SHA384']"
```

Comment:
If the cipher name changes, shouldn't we also increase our minimum version dependency?
Or how does this all fit together?

Reply:
Well.. the confusing part here is OpenSSL:
```python
ctx.set_cipher_list('AES256-GCM-SHA384')
ctx.get_cipher_list()  # returns ['AES256-GCM-SHA384', 'TLS_AES_256_GCM_SHA384', ...]
```
This is because they also introduced `SSL_set_ciphersuites` for TLS 1.3, and `SSL_get_cipher_list` returns whatever you set with `SSL_set_cipher_list` _plus_ what you set with `SSL_set_ciphersuites`. 🤷‍♂ 

I guess the correct approach for us would be to also introduce a cipher_suites option, and always set both. A lot of the code we're touching here will probably be thrown out with sans-io, so I decided to not fix all that up.

Minimum version dependency: cryptography is not guaranteed to ship with OpenSSL 1.1.1, you can also build it with older versions. So in either case we'd need to support both.